src/main/native/windows/process.cc

```diff
@@ -47,6 +51,81 @@ static bool DupeHandle(HANDLE h, AutoHandle* out, std::wstring* error) {
   return true;
 }
 
+static std::wstring CreateProcessWrapper(const std::wstring argv0,
+                                         const std::wstring &commandline,
+                                         std::wstring &out_path) {
+  // Safety first, ensure the out_path is empty before we do any work
+  out_path = L"";
+
+  /*
+    Save on possible collisions by using the argv0 name as a base for the
+    wrapper file name.
+    Argv0 is likely quoted, so we can use a string_view to cheaply manipulate
+    it and get rid of the quote if needed.
+  */
+  auto const view = std::wstring_view(argv0);
+  auto const pos = view.find_last_of('\\');
+  auto stem = view.substr(pos == std::wstring::npos ? 0 : (pos + 1));
+  if (stem.find(L"\"") != std::wstring::npos) {
+    /*
+     After removing the initial quote, there's no other place for the quote to
+     be than at the end, so we can simply remove it and move on.
+     */
+    stem.remove_suffix(1);
+  }
+
+  std::wstring wrapper_file_name = std::wstring(stem) + L".wrapper.bat";
```

Comment:
This might be risk of conflict with existing file or lack of write permission? Can we find a safer place to create the wrapper?